runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -67,6 +67,39 @@ public Layout getCurrentLayout() {
      */
     private AtomicReference<RuntimeLayout> runtimeLayout = new AtomicReference<>();
 
+    /**
+     * Fetches the layout uninterruptibly and rethrows any systemDownHandlerExceptions or Errors
+     * encountered.
+     *
+     * @return Fetched layout
+     */
+    private Layout getLayoutUninterruptibly() {
+        try {
+            return runtime.layout.get();
+        } catch (InterruptedException ie) {
+            throw new UnrecoverableCorfuInterruptedError("Interrupted in layoutHelper", ie);
+        } catch (ExecutionException ex) {
+            // Invalidate layout since the layout completable future now contains an exception.
+            runtime.invalidateLayout();
+            // If an error or an unchecked exception is thrown by the layout.get() completable
+            // future, the exception will materialize as an ExecutionException. In that case,
+            // we need to propagate this Error or unchecked exception.
+            if (ex.getCause() instanceof Error) {
+                log.error("getLayoutUninterruptibly: Encountered error. Aborting layoutHelper", ex);
+                throw (Error) ex.getCause();
+            }
+
+            if (ex.getCause() instanceof RuntimeException) {
+                log.error("getLayoutUninterruptibly: Encountered unchecked exception. "
+                        + "Aborting layoutHelper", ex);
+                throw (RuntimeException) ex.getCause();
+            }
+
+            log.error("getLayoutUninterruptibly: Encountered exception while fetching layout");
+            throw new RuntimeException(ex);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)